storage/keyring.go

```diff
@@ -335,3 +407,19 @@ func (kr *KeyringStore) DeleteRoleCredentials(arn string) error {
 func getHomePath(path string) string {
 	return strings.Replace(path, "~", os.Getenv("HOME"), 1)
 }
+
+func joinByteArray(chunks []([]byte)) []byte {
+	var result []byte
+	for _, s := range chunks {
+		result = append(result, s...)
+	}
+	return result
+}
+
+func getMyGOOS() string {
+	//For testing purposes..
+	if os.Getenv("GOOS_OVERRIDE") != "" {
```

Comment:
perhaps you meant to use https://pkg.go.dev/os#LookupEnv

Also, thinking about this more I'm a little worried that using an environment variable for this could have strange side effects during runtime if a user has this variable randomly set in their shell.  debugging that would probably be miserable.  Probably a good case for using a global var in the module which then could be overridden by your unit tests.  See `logger.go` for an example.

Reply:
I corrected in new version, now i'm using a global variable that can be set during tests:

```go
func (kr *KeyringStore) getMyGOOS() string {
	if myGOOS == "" {
		return runtime.GOOS
	}
	return myGOOS
}

func (kr *KeyringStore) setMyGOOS(os string) {
	myGOOS = os
}
```